src/toucan2/jdbc/result_set.clj

```diff
@@ -142,6 +145,7 @@
         col-names         (get builder :cols (next.jdbc.rs/get-modified-column-names
                                               (.getMetaData rset)
                                               combined-opts))
+        col-names-kw      (vec (keep #(column-name->keyword % label-fn) col-names))
         col-name->index   (make-column-name->index col-names label-fn)]
```

Comment:
Seems like it will be bad to performance to calculate all the column name keywords ahead of time in the line above and then create a memorized function that has to recalculate all those names a second time here... surely if you want to go down this road it would make sense to have `make-column-name->index` take `col-names-kw` so it doesn't need to call `column-name->keyword` again. Right?

Reply:
Nevermind I guess we still need it to work this way to power stuff like the underlying `IPersistentMap` methods

Reply:
Like I've said above, column names here are calculated only once per ResultSet. The current realized-keys update is triggered for every individual cell access (N columns * M rows). That's why optimizing the once per ResultSet part is not as important. But if there are easy wins there, it make sense to do them as well.

The reason why I left the current memoized `make-column-name->index` conversion is because, if I read the code correctly, the user may pass a string, a keyword, or a symbol as a column name, and they all should work. We can perhaps calculate a 3x map for all possible valid column names. I am not against it but it feels a bit dirty. We can go that route if you wish.

Reply:
No, scratch that. We can't precalculate all valid column names because of arbitrary `label-fn`. I see that it transforms names to kebab case for example, so `:my_good_column`, `"my-good_column"`, `'my_good-column` are all valid keys that should be transformed to `:my-good-coumn` in the end. We can't precalculate that.